doc/contributing.rst

```diff
@@ -0,0 +1,874 @@
+    with the following command::
+
+      pandoc doc/contributing.rst -t markdown_github > CONTRIBUTING.md
+
+
+How to build the *xarray* documentation
+---------------------------------------
+
+Requirements
+~~~~~~~~~~~~
+
+First, you need to have a development environment to be able to build xarray
+(see the docs on :ref:`creating a development environment above <contributing.dev_env>`).
+
+Building the documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+So how do you build the docs? Navigate to your local
+``xarray/doc/`` directory in the console and run::
+
+    make html
+
+Then you can find the HTML output in the folder ``xarray/doc/build/html/``.
+
+The first time you build the docs, it will take quite a while because it has to run
+all the code examples and build all the generated docstring pages. In subsequent
+evocations, sphinx will try to only build the pages that have been modified.
+
+If you want to do a full clean build, do::
+
+    make clean
+    make html
+
+.. _contributing.code:
+
+Contributing to the code base
+=============================
+
+.. contents:: Code Base:
+   :local:
+
+Code standards
+--------------
+
+Writing good code is not just about what you write. It is also about *how* you
+write it. During :ref:`Continuous Integration <contributing.ci>` testing, several
+tools will be run to check your code for stylistic errors.
+Generating any warnings will cause the test to fail.
+Thus, good style is a requirement for submitting code to *xarray*.
+
+In addition, because a lot of people use our library, it is important that we
+do not make sudden changes to the code that could have the potential to break
+a lot of user code as a result, that is, we need it to be as *backwards compatible*
+as possible to avoid mass breakages.
+
+Python (PEP8)
+~~~~~~~~~~~~~
+
+*xarray* uses the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_ standard.
+There are several tools to ensure you abide by this standard. Here are *some* of
+the more common ``PEP8`` issues:
+
+  - we restrict line-length to 79 characters to promote readability
+  - passing arguments should have spaces after commas, e.g. ``foo(arg1, arg2, kw1='bar')``
+
+:ref:`Continuous Integration <contributing.ci>` will run
+the `flake8 <http://pypi.python.org/pypi/flake8>`_ tool
+and report any stylistic errors in your code. Therefore, it is helpful before
+submitting code to run the check yourself on the diff::
+
+   git diff master -u -- "*.py" | flake8 --diff
+
+This command will catch any stylistic errors in your changes specifically, but
+be beware it may not catch all of them. For example, if you delete the only
+usage of an imported function, it is stylistically incorrect to import an
+unused function. However, style-checking the diff will not catch this because
+the actual import is not part of the diff. Thus, for completeness, you should
+run this command, though it will take longer::
+
+   git diff master --name-only -- "*.py" | grep "xarray/" | xargs -r flake8
```

Comment:
Running `flake8 xarray` takes 5.7s on my system, so that's may be the right slower-but-more-complete command